### PR TITLE
DOC-8391 - Server Compatibiliy - Transactions

### PR DESCRIPTION
**Pull request by ibsoln**
https://issues.couchbase.com/browse/DOC-8391

### CODE REVIEW

File: modules/ROOT/pages/server-compatibility-transactions.adoc [hunk 1]
```diff
@@ -3,4 +3,44 @@
...
+// BEGIN -- Page Heading
+:param-topic-group: compatibility
+:param-related: {compatibility--xref}
+include::partial$block-abstract.adoc[]
+// END -- Page Heading
+
+
+== Introduction
+
+Couchbase Server provides the backing data store for Sync Gateway and it is compatible -- see: {compatibility--xref}.
```
Comment (adamcfraser) [inline modules/ROOT/pages/server-compatibility-transactions.adoc]: "it is compatible" is an odd way to describe this - as you say, Couchbase Server is a dependency for Sync Gateway, so stating it's compatible with server (as a whole) doesn't really make sense.

File: modules/ROOT/pages/server-compatibility-transactions.adoc [hunk 2]
```diff
@@ -3,4 +3,44 @@
...
+However, there are considerations you needs to be aware of when using some Couchbase Server features with Sync Gateway.
+One of those instances concerns the use of Couchbase Server Transactions.
```
Comment (adamcfraser) [inline modules/ROOT/pages/server-compatibility-transactions.adoc]: This leaves me wondering what the other instances are.  In general I feel like this introduction could be rewritten along the lines of 'Couchbase mobile and server both support transactions, here are details on how they relate'

File: modules/ROOT/pages/server-compatibility-transactions.adoc [hunk 3]
```diff
@@ -3,4 +3,44 @@
...
+
+
+== Local Transactions
+
+Within the Couchbase Mobile cluster all transactions are local, either to the Couchbase Server or to the Couchbase Lite database:
```
Comment (adamcfraser) [inline modules/ROOT/pages/server-compatibility-transactions.adoc]: I don't think we really talk about a Couchbase Server+Mobile deployment as a Couchbase Mobile Cluster - at least that's not my usual understanding of a 'cluster'.  Maybe rephrase as 'Couchbase Mobile deployment' or 'Couchbase Mobile ecosystem'?

File: modules/ROOT/pages/server-compatibility-transactions.adoc [hunk 4]
```diff
@@ -3,4 +3,44 @@
...
+
+* Couchbase Server supports server side transactions -- see: {server-transactions--xref}. +
+Changes made on the server within a transaction block are guaranteed to be imported by the Sync Gateway only after all the changes in a transaction are committed.
+
+* Similarly, Couchbase Lite supports transactions that are local to the Couchbase Lite database -- see: {url-api-method-database-in-batch}. +
+Only committed transactions will ever reach Sync Gateway.
+
+Sync Gateway is not transaction-aware.
```
Comment (adamcfraser) [inline modules/ROOT/pages/server-compatibility-transactions.adoc]: You've already stated that transactions won't be imported by Sync Gateway until they are committed, so I don't think it's correct to say that Sync Gateway is not transaction aware.  I think the more important clarification is that mobile replication isn't transaction-aware, or more specifically doesn't replicate transactions atomically.  This isn't just a Sync Gateway consideration - Couchbase Lite doesn't replicate documents transactionally either (to SG, or P2P).

File: modules/ROOT/pages/server-compatibility-transactions.adoc [hunk 5]
```diff
@@ -3,4 +3,44 @@
...
+There is no guarantee that a series of updates made within a transaction block, on either the Couchbase Server or Couchbase Lite client side, will be ACID compliant when they sync to the other end.
+
+If there is a use-case to build end-to-end transactions between Couchbase Server and Couchbase Lite clients, you would implement it within the application layer.
+Perhaps by having a separate _control document_ of transaction log entries, which must be synced from the clients that capture the list of related changes.
```
Comment (adamcfraser) [inline modules/ROOT/pages/server-compatibility-transactions.adoc]: I don't know about proposing this hypothetical approach - wouldn't it be better to limit the docs to describing the functionality we do support?

**Review by adamcfraser — CHANGES_REQUESTED** [review]

File: modules/ROOT/pages/server-compatibility-transactions.adoc (hunk 1 above)
Comment (ibsoln) [reply modules/ROOT/pages/server-compatibility-transactions.adoc]: Agreed @adamcfraser  I will amend in line with comments before publishing.

File: modules/ROOT/pages/server-compatibility-transactions.adoc (hunk 2 above)
Comment (ibsoln) [reply modules/ROOT/pages/server-compatibility-transactions.adoc]: @adamcfraser  I will amend in line with comments before publishing.

File: modules/ROOT/pages/server-compatibility-transactions.adoc (hunk 3 above)
Comment (ibsoln) [reply modules/ROOT/pages/server-compatibility-transactions.adoc]: Ok. I'll amend ito ecosystem before publishing.

File: modules/ROOT/pages/server-compatibility-transactions.adoc (hunk 4 above)
Comment (ibsoln) [reply modules/ROOT/pages/server-compatibility-transactions.adoc]: @adamcfraser -- changed in line in published version..

File: modules/ROOT/pages/server-compatibility-transactions.adoc (hunk 5 above)
Comment (ibsoln) [reply modules/ROOT/pages/server-compatibility-transactions.adoc]: removed

**Comment by ibsoln** [comment]
Feedback applied prior to publishing
